docarray/array/storage/weaviate/find.py

```diff
@@ -27,26 +29,49 @@
 
 
 class FindMixin:
-    def _find_similar_vectors(self, query: 'WeaviateArrayType', limit=10):
+    def _find_similar_vectors(
+        self, query: 'WeaviateArrayType', limit=10, filter: Optional[Dict] = None
+    ):
         query = to_numpy_array(query)
         is_all_zero = np.all(query == 0)
         if is_all_zero:
             query = query + EPSILON
 
         query_dict = {'vector': query}
-        results = (
-            self._client.query.get(
-                self._class_name,
-                ['_serialized', '_additional {certainty}', '_additional {id}'],
+        if filter:
+            results = (
+                self._client.query.get(
+                    self._class_name,
+                    ['_serialized', '_additional {certainty}', '_additional {id}'],
+                )
+                .with_where(filter)
+                .with_limit(limit)
+                .with_near_vector(query_dict)
+                .do()
+            )
+        else:
+            results = (
+                self._client.query.get(
+                    self._class_name,
+                    ['_serialized', '_additional {certainty}', '_additional {id}'],
+                )
+                .with_limit(limit)
+                .with_near_vector(query_dict)
+                .do()
             )
-            .with_limit(limit)
-            .with_near_vector(query_dict)
-            .do()
-        )
         docs = []
 
+        found_results = (
+            results.get('data', {}).get('Get', {}).get(self._class_name, []) or []
+        )
+        if 'errors' in results:
```

Comment:
it is weird to do this check after extracting from `found_results`, shouldn't it be done before?

Reply:
fixed